python/lsst/multiprofit/componentconfig.py

```diff
@@ -0,0 +1,469 @@
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = [
+    "ParameterConfig",
+    "FluxFractionParameterConfig",
+    "FluxParameterConfig",
+    "CentroidConfig",
+    "ComponentData",
+    "Fluxes",
+    "EllipticalComponentConfig",
+    "GaussianComponentConfig",
+    "SersicIndexParameterConfig",
+    "SersicComponentConfig",
+]
+
+import string
+
+import lsst.gauss2d.fit as g2f
+import lsst.pex.config as pexConfig
+import pydantic
+
+from .limits import limits_ref
+from .priors import ShapePriorConfig
+from .transforms import transforms_ref
+from .utils import frozen_arbitrary_allowed_config
+
+
+class ParameterConfig(pexConfig.Config):
+    """Configuration for a parameter."""
+
+    fixed = pexConfig.Field[bool](default=False, doc="Whether parameter is fixed or not (free)")
+    value_initial = pexConfig.Field[float](default=0, doc="Initial value")
+
+
+class FluxParameterConfig(ParameterConfig):
+    """Configuration for flux parameters (IntegralParameterD).
+
+    The safest initial value for a flux is 1.0, because if it's set to zero,
+    linear fitting will not work correctly initially.
+    """
+
+    def setDefaults(self):
+        super().setDefaults()
+        self.value_initial = 1.0
+
+
+class FluxFractionParameterConfig(ParameterConfig):
+    """Configuration for flux fraction parameters (ProperFractionParameterD).
+
+    The safest initial value for a flux fraction is 0.5, because if it's set
+    to one, downstream fractions will be zero, while if it's set to zero,
+    linear fitting will not work correctly initially.
+    """
+
+    def setDefaults(self):
+        super().setDefaults()
+        self.value_initial = 1.0
+
+
+class CentroidConfig(pexConfig.Config):
+    """Configuration for a component centroid."""
+
+    x = pexConfig.ConfigField[ParameterConfig](doc="The x-axis centroid configuration")
+    y = pexConfig.ConfigField[ParameterConfig](doc="The y-axis centroid configuration")
+
+    def make_centroid(self) -> g2f.CentroidParameters:
+        cen_x, cen_y = (
+            type_param(config.value_initial, fixed=config.fixed, limits=g2f.LimitsD())
+            for (config, type_param) in ((self.x, g2f.CentroidXParameterD), (self.y, g2f.CentroidYParameterD))
+        )
+        centroid = g2f.CentroidParameters(x=cen_x, y=cen_y)
+        return centroid
```

Comment:
I think this would be more readable as
```suggestion
    def make_centroid(self) -> g2f.CentroidParameters:
        cen_x = g2f.CentroidXParameterD(self.x.value_initial, fixed=self.x.fixed, limits=g2f.LimitsD())
        cen_y = g2f.CentroidYParameterD(self.y.value_initial, fixed=self.y.fixed, limits=g2f.LimitsD())
        centroid = g2f.CentroidParameters(x=cen_x, y=cen_y)
        return centroid
```

Reply:
I agree, but this just barely met my threshold for adding a loop to avoid accidentally referencing the wrong corresponding object.